improver_tests/nbhood/recursive_filter/test_RecursiveFilter.py

```diff
@@ -352,50 +352,51 @@ def test_result_basic(self):
-        )
-        smoothing_coefficients_y = RecursiveFilter()._set_smoothing_coefficients(
-            cube, self.smoothing_coefficient_y, None
-        )
+        smoothing_coefficients_x = RecursiveFilter(
+            edge_width=edge_width
+        )._set_smoothing_coefficients(cube, self.smoothing_coefficient_x, None)
+        smoothing_coefficients_y = RecursiveFilter(
+            edge_width=edge_width
+        )._set_smoothing_coefficients(cube, self.smoothing_coefficient_y, None)
         padded_cube = pad_cube_with_halo(cube, 2 * edge_width, 2 * edge_width)
-        result = RecursiveFilter()._run_recursion(
+        result = RecursiveFilter(edge_width=edge_width)._run_recursion(
             padded_cube,
             smoothing_coefficients_x,
             smoothing_coefficients_y,
             self.iterations,
         )
-        expected_result = 0.13382206
+        expected_result = 0.12302627
         self.assertAlmostEqual(result.data[4][4], expected_result)
 
     def test_different_smoothing_coefficients(self):
         """Test that the _run_recursion method returns expected values when
         smoothing_coefficient values are different in the x and y directions"""
+        edge_width = 1
         cube = iris.util.squeeze(self.cube)
         smoothing_coefficient_y = 0.5 * self.smoothing_coefficient_x
-        smoothing_coefficients_x = RecursiveFilter()._set_smoothing_coefficients(
-            cube, self.smoothing_coefficient_x, None
-        )
-        smoothing_coefficients_y = RecursiveFilter()._set_smoothing_coefficients(
-            cube, smoothing_coefficient_y, None
-        )
-        padded_cube = pad_cube_with_halo(cube, 2, 2)
-        result = RecursiveFilter()._run_recursion(
+        smoothing_coefficients_x = RecursiveFilter(
+            edge_width=edge_width
+        )._set_smoothing_coefficients(cube, self.smoothing_coefficient_x, None)
+        smoothing_coefficients_y = RecursiveFilter(
+            edge_width=edge_width
+        )._set_smoothing_coefficients(cube, smoothing_coefficient_y, None)
+        padded_cube = pad_cube_with_halo(cube, 2 * edge_width, 2 * edge_width)
+        result = RecursiveFilter(edge_width=edge_width)._run_recursion(
             padded_cube, smoothing_coefficients_x, smoothing_coefficients_y, 1
         )
         # slice back down to the source grid - easier to visualise!
         unpadded_result = result.data[2:-2, 2:-2]
 
         expected_result = np.array(
             [
-                [0.01620921, 0.02866841, 0.05077430, 0.02881413, 0.01657352],
-                [0.03978802, 0.06457599, 0.10290188, 0.06486591, 0.04051282],
-                [0.10592333, 0.15184643, 0.19869247, 0.15238355, 0.10726611],
-                [0.03978982, 0.06457873, 0.10290585, 0.06486866, 0.04051464],
-                [0.01621686, 0.02868005, 0.05079120, 0.02882582, 0.01658128],
-            ]
+                [0.01320939, 0.02454378, 0.04346254, 0.02469828, 0.01359563],
```

Comment:
Without doing quite a bit of maths it is hard to say if these numbers are correct, but I don't see any reason why they shouldn't be.